base/task.jl

```diff
@@ -197,6 +215,7 @@ function produce(v)
     t.state == :runnable || throw(AssertionError("producer.consumer.state == :runnable"))
     if empty
         schedule_and_wait(t, v)
+        ct = current_task() # When a task is copied, ct should be updated to new task ID.
```

Comment:
And as I said in the old PR, I don't think we should allow copying of running task.